room/room.go

```diff
@@ -90,6 +92,36 @@ func (t *TalkRoom) SendMessage(msg string) (*ocs.TalkRoomMessageData, error) {
 	return &msgInfo.OCS.TalkRoomMessage, err
 }
 
+// DeleteMessage deletes the message with the given messageID on the server.
+//
+// Requires "delete-messages" capability on the Nextcloud Talk server
+func (t *TalkRoom) DeleteMessage(messageID int) (*ocs.TalkRoomMessageData, error) {
+	// Check for required capability
+	capable, err := t.User.Capabilities()
+	if err != nil {
+		return nil, err
+	}
+	if !capable.DeleteMessages {
+		return nil, ErrLackingCapabilities
+	}
+
+	url := t.User.NextcloudURL + constants.BaseEndpoint + "/chat/" + t.Token + "/" + strconv.Itoa(messageID)
+
+	client := t.User.RequestClient(request.Client{
+		URL:    url,
+		Method: "DELETE",
```

Comment:
So this now tries to delete a message in Talk?
That will only work within 6h of posting and only if you are the author or a moderator, so won't work with the token maybe?

I mostly ment the other way around, so when a Message is deleted in Talk, it should ping the other bridges to delete stuff too?

Reply:
This goes with https://github.com/42wim/matterbridge/pull/1492 on the Matterbridge side which will forward deletions in both directions. [Talk -> Others](https://github.com/42wim/matterbridge/pull/1492/files#diff-dc7ac4842d7eb8f457c012ed8b60abfa3861e627dd9471035ce8b23454c565cfR93-R107) and [Others -> Talk](https://github.com/42wim/matterbridge/pull/1492/files#diff-dc7ac4842d7eb8f457c012ed8b60abfa3861e627dd9471035ce8b23454c565cfR170-R180).

If it fails to delete, that will be in the debugging info but otherwise will have no other effect.